src/dist/component/package.rs

```diff
@@ -193,10 +193,15 @@ fn unpack_ram(
         }
     };
 
-    if io_chunk_size > unpack_ram {
-        panic!("RUSTUP_UNPACK_RAM must be larger than {}", io_chunk_size);
+    if minimum_ram > unpack_ram {
+        warn!(
+            "Available/permitted unpack RAM ({}) less than {}.  Unpacking may fail.",
+            unpack_ram, minimum_ram
```

Comment:
So unpack_ram may be low for several reasons.

It may be low due to environment variable overrides, in which case, we'll probably succeed in allocating minimum_ram and really what we're doing is ignoring a bad variable.

It may be low because we correctly detected a low available virtual address space / ulimit, in which case allocation of RAM will fail, and we will panic, and what we're doing is walking into trouble.

I think the right thing to do is not to override here, but to emit a warning and ignore higher up, on the specific branch that consults the environment variable. Let the low detected limit be actual.

Reply:
I concur, and since I'm in there, I may as well resolve the note about warning if the total is above detected RAM limits too.